go/test/endtoend/vtgate/queries/foundrows/found_rows_test.go

```diff
@@ -58,3 +64,38 @@ func TestFoundRows(t *testing.T) {
 	utils.Exec(t, conn, "delete from t2")
 	utils.Exec(t, conn, "delete from t2_id4_idx")
 }
+
+// waitForT2Authoritative waits for the table t2 to become authoritative
+func waitForT2Authoritative(t *testing.T) error {
+	timeout := time.After(10 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("schema tracking didn't mark table t2 as authoritative until timeout")
+		default:
+			time.Sleep(1 * time.Second)
+			res, err := clusterInstance.VtgateProcess.ReadVSchema()
+			require.NoError(t, err, res)
+			t2Map := getTableT2Map(res)
+			authoritative, fieldPresent := t2Map["column_list_authoritative"]
+			if !fieldPresent {
+				continue
+			}
+			authoritativeBool, isBool := authoritative.(bool)
+			if !isBool || !authoritativeBool {
+				continue
+			}
+			return nil
+		}
+	}
+}
+
+func getTableT2Map(res *interface{}) map[string]interface{} {
+	t2Map := convertToMap(convertToMap(convertToMap(convertToMap(convertToMap(*res)["keyspaces"])[KeyspaceName])["tables"])["t2"])
+	return t2Map
+}
+
+func convertToMap(input interface{}) map[string]interface{} {
+	output, _ := input.(map[string]interface{})
```

Comment:
nit: if you don't care about the boolean, you can just omit it altogether:

```
output := input.(map[string]interface{})
```

Reply:
Okay 👍